api/v1alpha1/kepler_x_types.go

```diff
@@ -0,0 +1,176 @@
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:validation:Minimum=1
+	Port int32 `json:"port,omitempty"`
+
+	// Defines which Nodes the Pod is scheduled on
+	// +optional
+	// +kubebuilder:default={"kubernetes.io/os":"linux"}
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, define Pod's tolerations
+	// +optional
+	// +kubebuilder:default={{"key": "", "operator": "Exists", "value": "", "effect": ""}}
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type KeplerXConfigSpec struct {
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Maximum=500
+	// +kubebuilder:validation:Minimum=1
+	MonitorInterval int32 `json:"monitorInterval,omitempty"`
+
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Maximum=500
+	// +kubebuilder:validation:Minimum=1
+	ExporterInterval int32 `json:"exporterInterval,omitempty"`
+
+	// +kubebuilder:default=100
+	// +kubebuilder:validation:Minimum=0
+	MaxTrackedTerminatedProcesses int32 `json:"maxTrackedTerminatedProcesses,omitempty"`
+
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Minimum=1
+	ProcessRetentionPeriod int32 `json:"processRetentionPeriod,omitempty"`
+}
+
+type KeplerXDashboardSpec struct {
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+type KeplerXOpenShiftSpec struct {
+	// +kubebuilder:default=true
+	Enabled   bool                 `json:"enabled"`
+	Dashboard KeplerXDashboardSpec `json:"dashboard,omitempty"`
+}
+
+// KeplerXSpec defines the desired state of KeplerX
+type KeplerXSpec struct {
+	// +kubebuilder:validation:Required
+	Deployment    KeplerXDeploymentSpec `json:"deployment"`
+	Configuration KeplerXConfigSpec     `json:"configuration,omitempty"`
+	OpenShift     KeplerXOpenShiftSpec  `json:"openshift,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:scope="Cluster"
+//+kubebuilder:subresource:status
+
+// +kubebuilder:printcolumn:name="Port",type=integer,JSONPath=`.spec.deployment.port`
+// +kubebuilder:printcolumn:name="Desired",type=integer,JSONPath=`.status.desiredNumberScheduled`
+// +kubebuilder:printcolumn:name="Current",type=integer,JSONPath=`.status.currentNumberScheduled`
+// +kubebuilder:printcolumn:name="Up-to-date",type=integer,JSONPath=`.status.updatedNumberScheduled`
+// +kubebuilder:printcolumn:name="Ready",type=integer,JSONPath=`.status.numberReady`
+// +kubebuilder:printcolumn:name="Available",type=integer,JSONPath=`.status.numberAvailable`
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:printcolumn:name="Image",type=string,JSONPath=`.spec.deployment.image`
+// +kubebuilder:printcolumn:name="Node-Selector",type=string,JSONPath=`.spec.deployment.nodeSelector`,priority=10
+// +kubebuilder:printcolumn:name="Tolerations",type=string,JSONPath=`.spec.deployment.tolerations`,priority=10
+//
+// KeplerX is the Schema for the internal kepler 2 API
+type KeplerX struct {
```

Comment:
Just a thought, how about we call it PowerMonitor (just like you had it earlier) ... 😬